cpp/oneapi/dal/algo/knn/backend/gpu/infer_kernel_impl_dpc_distr.hpp

```diff
@@ -699,26 +647,28 @@ sycl::event bf_kernel_distr(sycl::queue& queue,
             next_event = search(query, callback, qbcount, curr_k, { next_event });
         }
 
-        auto send_count = current_block.get_count();
-        ONEDAL_ASSERT(send_count >= 0);
-        ONEDAL_ASSERT(send_count <= de::limits<int>::max());
-        // send recv replace
-        comm.sendrecv_replace(array<Float>::wrap(queue,
-                                                 current_block.get_mutable_data(),
-                                                 send_count,
-                                                 { next_event }),
-                              prev_node,
-                              next_node)
-            .wait();
-        train_block_queue.emplace_back(current_block);
-        comm.sendrecv_replace(array<res_t>::wrap(queue,
-                                                 current_tresps.get_mutable_data(),
-                                                 current_tresps.get_count(),
-                                                 { next_event }),
-                              prev_node,
-                              next_node)
-            .wait();
-        tresps_queue.emplace_back(current_tresps);
+        if (relative_block_idx < block_count - 1) {
+            ONEDAL_PROFILER_TASK(distributed_loop.sendrecv_replace, queue);
+            auto send_count = current_block.get_count();
+            ONEDAL_ASSERT(send_count >= 0);
+            ONEDAL_ASSERT(send_count <= de::limits<int>::max());
+            comm.sendrecv_replace(array<Float>::wrap(queue,
+                                                     current_block.get_mutable_data(),
+                                                     send_count,
+                                                     { next_event }),
```

Comment:
I highly encourage you to use separate variables for the data:
```suggestion
            auto current_block_array = array<Float>::wrap(queue,
                                                     current_block.get_mutable_data(),
                                                     send_count,
                                                     { next_event });
            comm.sendrecv_replace(current_block_array,
```